components/reusable-components/blogDetails-card/BlogDetailsCard.tsx

```diff
@@ -0,0 +1,305 @@
+      { id: '2', first_name: 'Jane', last_name: 'Smith' },
+      { id: '3', first_name: 'Emily', last_name: 'Johnson' },
+    ],
+    []
+  );
+
+  const [isEditable, setIsEditable] = useState<boolean>(false);
+  const [selectedAuthorId, setSelectedAuthorId] = useState<string>('');
+  const [imageFile, setImageFile] = useState<File | null>(null);
+  const [validationErrors, setValidationErrors] = useState<{ image: string | null }>({
+    image: null,
+  });
+
+  const {
+    getRootProps,
+    getInputProps,
+    acceptedFiles,
+    fileRejections: rejectedFiles,
+  } = useDropzone({
+    accept: {
+      'image/*': [],
+    },
+    maxSize: 5 * 1024 * 1024,
+    onDrop: (files) => {
+      if (files.length > 0) {
+        setImageFile(files[0]);
+        onImageChange(files);
+      }
+    },
+    onDropRejected: (rejectedFiles) => {
+      if (process.env.NODE_ENV !== 'production') {
+        console.log('File Rejections:', rejectedFiles);
+      }
+    },
+  });
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const editMode = searchParams.get('edit') === 'true';
+    setIsEditable(editMode);
+  }, [searchParams]);
+
+  useEffect(() => {
+    const authorMatch = hardcodedAuthors.find(
+      (a) => a.first_name === propAuthor.first_name && a.last_name === propAuthor.last_name
+    );
+    if (authorMatch) {
+      setSelectedAuthorId(authorMatch.id);
+    }
+  }, [propAuthor, hardcodedAuthors]);
+
+  const handleEditClick = () => {
+    const form = document.querySelector('form') as HTMLFormElement;
+    const isImageRequired = isEditable && !imageFile && !imageUrl;
+
+    if (form && form.checkValidity() && !isImageRequired) {
+      setIsEditable((prevEditable) => !prevEditable);
+      setValidationErrors({ image: null });
+    } else {
+      form?.reportValidity();
+      if (isImageRequired) {
+        setValidationErrors({ image: 'Image is required.' });
+      }
+    }
+  };
+
+  const handleCancelClick = () => {
+    setIsEditable(false);
+    router.replace(window.location.pathname);
+    if (onCancelClick) {
+      onCancelClick();
+    }
+  };
+
+  const handleAuthorChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedAuthor = hardcodedAuthors.find((a) => a.id === event.target.value);
+    if (selectedAuthor) {
+      onChange({
+        target: {
+          name: 'author',
+          value: `${selectedAuthor.first_name} ${selectedAuthor.last_name}`,
+        },
+      });
+      setSelectedAuthorId(selectedAuthor.id);
+    }
+  };
+
+  return (
+    <form className={styles.blogDetailsCard} onSubmit={(e) => e.preventDefault()}>
+      <div className={styles.actionButtons}>
+        <div className={styles.leftButton}>
+          <button
+            type="button"
+            onClick={() => router.push('/content-panel/blogs')}
+            aria-label="Go back"
+          >
+            <i className="bi bi-arrow-left" />
+          </button>
```

Comment:
If the user has started editing and clicks Back, we should signal to the user through the UI that the changes that were made are going to be discarded.
On click on Back you should show a modal.